autokeras/auto_model.py

```diff
@@ -87,15 +95,15 @@ class AutoModel(object):
     """
 
     def __init__(self,
-                 inputs,
-                 outputs,
-                 name='auto_model',
-                 max_trials=100,
-                 directory=None,
-                 objective='val_loss',
-                 tuner='greedy',
-                 overwrite=False,
-                 seed=None):
+                 inputs: Union[Input, List[Input]],
+                 outputs: Union[head_module.Head, list],
```

Comment:
It should be `Union[Head, Node, list]`. I think you accidentally removed `Node`.
Another concern is how will it show on the docs by autogen.py?
Will it be `head_module.Head` directly? That would be confusing for the reader.

Reply:
Thanks, I'll fix it.

Good question, I'll dig into it. I'm planning to change the code generation too to display the types in the description, not the signature as it's too cumbersome and hurts readability.

Reply:
Here is what it looks like in the docs: 

```
autokeras.AutoModel(
    inputs: Union[autokeras.nodes.Input, List[autokeras.nodes.Input]],
    outputs: Union[autokeras.engine.head.Head, list],
    name: str = "auto_model",
    max_trials: int = 100,
    directory: Union[str, pathlib.Path, None] = None,
    objective: str = "val_loss",
    tuner: Union[str, Type[autokeras.engine.tuner.AutoTuner]] = "greedy",
    overwrite: bool = False,
    seed: Optional[int] = None,
)
```

I can see two things to fix:

* Fist, use the aliases in the type hints instead of the complete class path.
* Second, display those info in the arguments descriptions only.

I'll open two issue in keras-autodoc. I'll fix the aliases issue first, and I believe we can merge it once it's done to avoid blocking the typing of autokeras.

The second issue is more or less making the docs beautiful, which is important, but not critical.

Reply:
It looks like that now: 

```python
autokeras.AutoModel(
    inputs: Union[autokeras.Input, List[autokeras.Input]],
    outputs: Union[autokeras.Head, autokeras.Node, list],
    name: str = "auto_model",
    max_trials: int = 100,
    directory: Union[str, pathlib.Path, None] = None,
    objective: str = "val_loss",
    tuner: Union[str, Type[autokeras.engine.tuner.AutoTuner]] = "greedy",
    overwrite: bool = False,
    seed: Optional[int] = None,
)
```

`AutoTuner` is not listed in the docs, so there is no aliases, this is why keras-autodoc can't use an alias for it.

Do we plan to make `AutoTuner` public in the future?

Reply:
Yes, we should make it public for anyone would like to override to implement their own tuner. I will merge this PR now.

Would you also help us make typedapi part of the setup.py tests dependencies? That would make the community easier to setup the environment. Thank you!

Reply:
Sure thing, I'll upload it on pypi when I have the time :)